src/abs.ott

```diff
@@ -108,32 +108,30 @@ formula :: formula_ ::=
     {{ coq (Map.find [[x]] [[s]] = Some ([[t]])) }}
   | G ( fn ) = sig :: M :: G_fn_eq_sig
     {{ coq (Map.find [[fn]] [[G]] = Some (ctxv_sig [[sig]])) }}
-  | fresh ( x1 , ... , xn , e ) :: M :: fresh
+  | fresh ( x1 , ... , xn , e ) :: M :: fresh_e
     {{ coq (fresh_vars_e [[x1 ... xn]] [[e]]) }}
+  | fresh ( x1 , ... , xn , s ) :: M :: fresh_s
+    {{ coq (fresh_vars_s [[x1 ... xn]] [[s]]) }}
+  | fresh ( x1 , ... , xn , e , s ) :: M :: fresh
+    {{ coq (fresh_vars [[x1 ... xn]] [[e]] [[s]]) }}
+  | well_formed ( x1 , ... , xn , e , s ) :: M :: well_formed
+    {{ coq (well_formed [[e]] [[s]] [[x1 ... xn]]) }}
 
 embed
 {{ coq
-Fixpoint get_val (k:x) (l:list(x*x)): option x :=
-  match l with
-  | nil => None
-  | (k', v) :: l' =>
-      match eq_x k k' with
-      | left _ => Some v
-      | _ => get_val k l'
-      end
-  end.
+Equations e_var_subst_one (e5:e) (x_ y_: x) : e := {
+ e_var_subst_one (e_t t) _ _ := e_t t;
+ e_var_subst_one (e_var x0) x_ y_ := if (eq_x x0 x_) then (e_var y_) else (e_var x0);
+ e_var_subst_one (e_fn_call fn0 arg_list) x_ y_ := e_fn_call fn0 (e_list_subst_one arg_list x_ y_) }
+where e_list_subst_one (es:list e) (x_ y_: x) : list e := {
+ e_list_subst_one nil _ _ := nil;
+ e_list_subst_one (e0::es) x_ y_ := e_var_subst_one e0 x_ y_ :: e_list_subst_one es x_ y_
+}.
 
-Fixpoint e_var_subst (e5:e) (l:list (x*x)) : e :=
-  match e5 with
-  | e_t t => e_t t
-  | e_var x =>
-      match get_val x l with
-              | None => e_var x
-              | Some y => e_var y
-              end
-  | e_fn_call fn e_list =>
-      e_fn_call fn (map (fun e => e_var_subst e l) e_list)
-  end.
+(* I would prefer not doing this, but sometimes I need it to compute *)
+Global Transparent e_var_subst_one.
```

Comment:
There is also the option `Set Equations Transparent.`, but if this is the only definition that needs transparency, I guess it works.